File: lib/source/tagset.cpp
#include "tagset.hpp"

using namespace snow;

Tagset::~Tagset(){
	std::vector<Flake*>::iterator it=tagnames.begin(),end=tagnames.end();
	for(;it!=end;++it){
		delete *it;
	}
}

Tagdef Tagset::get(Flake* name){
	std::vector<Flake*>::iterator it=
		tagnames.begin(),
		end=tagnames.end(),
		start=it;

	for(;it!=end;++it){
		if(name->eq(*it)){
			return defs[it-start];
		}
	}

	return Tagdef();
}

bool Tagset::add(Flake* name,Tagdef def){
	if(get(name)){
		return false;
	}

	tagnames.push_back(name);
	defs.push_back(def);
	return true;
}

File: java/Tagset.java
package org.snowlang.snow;

import java.util.List;
import java.util.Arrays;

public class Tagset{
	protected List<Tagdef> defs;
	
	public Tagset(Tagdef... d){
		defs=Arrays.asList(d);
	}
	
	public Tagdef get(int x){
		return defs.get(x);
	}
}